atlasdb-ete-tests/build.gradle

```diff
@@ -94,50 +94,111 @@ dependencies {
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.MultiClientWithPostgresTimelockAndPostgresTestSuite'
+    }
 }
 
-task dbkvsTest(type: Test) {
-    dependsOn prepareForEteTests
-    include '**/DbKvsTestSuite.class'
+task multiClientWithTimelockAndCassandraTest(type: Test, dependsOn: 'prepareForTimelockEteTests') {
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.MultiClientWithTimelockAndCassandraTestSuite'
+    }
 }
 
-task oracleTest(type: Test) {
-    dependsOn prepareForEteTests
-    include '**/OracleDbKvsEteTestSuite.class'
+task singleClientWithEmbeddedAndCassandraTest(type: Test, dependsOn: 'prepareForEteTests') {
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndCassandraTestSuite'
+    }
 }
 
-test {
-    dependsOn longTest, prepareForEteTests, timeLockTest, dbkvsTest
-    exclude '**/MultiCassandraTestSuite.class'
+task singleClientWithEmbeddedAndOracleTest(type: Test, dependsOn: 'prepareForEteTests') {
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndOracleTestSuite'
+    }
+}
 
-    exclude '**/*TimeLock*.class'
-    // Technically covered by previous line, but explicitly avoiding in case of refactor
-    exclude '**/*TimeLockMigration*.class'
+task singleClientWithEmbeddedAndPostgresTest(type: Test, dependsOn: 'prepareForEteTests') {
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndPostgresTestSuite'
+    }
+}
+
+task singleClientWithEmbeddedAndThreeNodeCassandraTest(type: Test, dependsOn: 'prepareForEteTests') {
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndThreeNodeCassandraTestSuite'
+    }
+}
+
+// Re-wire 'test' to include custom tasks as dependencies via a single task instead of directly running them
 
-    exclude '**/*EteTest.class'
-    exclude '**/DbKvsTestSuite.class'
-    exclude '**/OracleDbKvsEteTestSuite.class'
+task customEteTasks() {
+    dependsOn multiClientWithPostgresTimelockAndPostgresTest,
+        multiClientWithTimelockAndCassandraTest,
+        singleClientWithEmbeddedAndCassandraTest,
+        singleClientWithEmbeddedAndOracleTest,
+        singleClientWithEmbeddedAndPostgresTest,
+        singleClientWithEmbeddedAndThreeNodeCassandraTest
+        // timeLockMigrationTest // intentionally not run on every commit/PR due to length
+}
+
+test {
+    dependsOn customEteTasks
+
+    filter {
+        // in the expected case "test" doesn't actually have any tests to run because they're all in custom tasks
+        // gradle would normally fail a task of type Test that doesn't actually have any tests to run
+        setFailOnNoMatchingTests(false)
+
+        // suite classes that should not be run outside of suites
+        excludeTestsMatching 'com.palantir.atlasdb.ete.suiteclasses.*'
```

Comment:
Just for my understanding, this is necessary because we're excluding the suites that would otherwise be responsible for running these tests?

Reply:
Correct. This was happening before this PR via the `exclude '**/*EteTest.class'` line. If they run on their own they immediately fail because there's no clients actually running (that would otherwise be spun up by the Suite's ClassRule).